net/nimble/controller/src/ble_ll_scan.c

```diff
@@ -912,6 +913,13 @@ ble_ll_scan_start(struct ble_ll_scan_sm *scansm)
     int phy;
 
     ble_ll_get_chan_to_scan(scansm, &scan_chan, &phy);
+
+    /* XXX: right now scheduled item is only present if we schedule for aux
+     *      scan just make sanity check that we have proper combination of
+     *      sch and resulting scan_chan
+     */
+    assert(!sch || scan_chan < BLE_PHY_ADV_CHAN_START);
+    assert(sch || scan_chan >= BLE_PHY_ADV_CHAN_START);
```

Comment:
I'd make those assert like this:
assert(sch && scan_chan < BLE_PHY_ADV_CHAN_START);
assert(!sch && scan_chan >= BLE_PHY_ADV_CHAN_START);